.github/workflows/android-build.yml

```diff
@@ -11,13 +11,15 @@ on:
   push:
     branches:
       - Reanimated2
+
 jobs:
   build:
     runs-on: ubuntu-latest
-    env:
-      WORKING_DIRECTORY: Example
+    strategy:
+      matrix:
+        working-directory: [Example]
     concurrency:
-      group: android-${{ github.ref }}
+      group: rea2-android-${{ matrix.working-directory }}-${{ github.ref }}
```

Comment:
It looks like `github.ref` is sufficient to distinguish v3 from v2 as it contains the branch name or the number of PR, which is bound to a specific version, so we can remove `rea2-` here.